File: app/src/main/java/agersant/polaris/api/remote/APIVersion5.java
package agersant.polaris.api.remote;

import android.net.Uri;

import agersant.polaris.api.ThumbnailSize;

class APIVersion5 extends APIVersion4 {
    APIVersion5(DownloadQueue downloadQueue, RequestQueue requestQueue) {
        super(downloadQueue, requestQueue);
    }

    @Override
    String getAudioURL(String path) {
        String serverAddress = ServerAPI.getAPIRootURL();
        return serverAddress + "/audio/" + Uri.encode(path);
    }

    @Override
    String getThumbnailURL(String path, ThumbnailSize size) {
        String serverAddress = ServerAPI.getAPIRootURL();
        return serverAddress + "/thumbnail/" + Uri.encode(path);
    }
}

File: app/src/main/java/agersant/polaris/api/remote/APIVersion5.kt
package agersant.polaris.api.remote

import agersant.polaris.api.ThumbnailSize
import android.net.Uri
import io.ktor.client.*

internal open class APIVersion5(
    downloadQueue: DownloadQueue,
    client: HttpClient,
    apiRootUrl: String,
) : APIVersion4(downloadQueue, client, apiRootUrl) {

    override fun getAudioUrl(path: String): String {
        return "$apiRootUrl/audio/${Uri.encode(path)}"
    }

    override fun getThumbnailUrl(path: String, size: ThumbnailSize): String {
        return "$apiRootUrl/thumbnail/${Uri.encode(path)}"
    }
}
